addon/Simulator.jsm

```diff
@@ -0,0 +1,141 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+"use strict";
+
+const {classes:Cc, interfaces: Ci, utils: Cu} = Components;
+
+const EXPORTED_SYMBOLS = ["Simulator"];
+
+Cu.import("resource://gre/modules/Services.jsm");
+Cu.import("resource://gre/modules/Console.jsm");
+Cu.import("resource://gre/modules/FileUtils.jsm");
+Cu.import("resource://gre/modules/Log.jsm");
+
+const PREF_HOMEPAGE = "browser.startup.homepage";
+
+const TELEMETRY_RAPPOR_PATH = `chrome://shield-study-rappor/content/TelemetryRappor.jsm`;
+const { TelemetryRappor } = Cu.import(TELEMETRY_RAPPOR_PATH, {});
+
+const log = createLog("Simulator", "Info");
+
+/**
+ * Create the logger
+ * @param {string} name - Name to show in the logs.
+ * @param {string} level - Log level.
+ */
+function createLog(name, level) {
+  var logger = Log.repository.getLogger(name);
+  logger.level = Log.Level[level] || Log.Level.Debug;
+  logger.addAppender(new Log.ConsoleAppender(new Log.BasicFormatter()));
+  return logger;
+}
+
+/**
+ * Runs the simulation and writes the data into case_reports.csv.
+ * For running the simulation, the true values from case_true_values.csv
+ * are read, and RAPPOR is executed.
+ * @param {string} studyName - Name of the study.
+ * @param {string} rapporPath - Path where the RAPPOR simulator lives.
+ * @param {object} params - Object containing the algorithm parameters.\
+ * @param {constant} method  - Method used to encode. Supported: nsICryptoHash.MD5 and nsICryptoHash.SHA256
+ * @param {string} instance - Simulation instance.
+ */
+function runRapporSimulation(studyName, rapporPath, params, method, instance) {
+  let data = read(new FileUtils.File(rapporPath + "_tmp/python/" + instance + "/1/case_true_values.csv"));
+  let caseReportsFile = new FileUtils.File(rapporPath + "_tmp/python/" + instance + "/1/case_reports.csv");
+  write(caseReportsFile, "client, cohort, bloom, prr, irr\n");
+  // iterate over each line of the file getting the
+  // client, cohort and value.
+  for (let i = 1; i < data.length; i++) {
+    let line = data[i].split(",");
+    let report = TelemetryRappor.createReport(studyName, line[2], params, method, true, line[1]);
+    write(caseReportsFile,
```

Comment:
Please document the expected format of the output file in this function.

Reply:
Done